tests/unit/Imports/ProductImportTest.php

```diff
@@ -0,0 +1,134 @@
+
+use Mockery\MockInterface;
+use StoreKeeper\WooCommerce\B2C\Commands\ProcessAllTasks;
+use StoreKeeper\WooCommerce\B2C\Imports\AbstractProductImport;
+use StoreKeeper\WooCommerce\B2C\Options\StoreKeeperOptions;
+use StoreKeeper\WooCommerce\B2C\Tools\StoreKeeperApi;
+use StoreKeeper\WooCommerce\B2C\UnitTest\Commands\AbstractTest;
+use Throwable;
+
+class ProductImportTest extends AbstractTest
+{
+    public const MEDIA_DATADUMP_DIRECTORY = 'imports/products/media';
+
+    public const CREATE_DATADUMP_SUCCESS_HOOK = 'imports/hook.events.createProduct.success.json';
+    public const CREATE_DATADUMP_FAILED_HOOK = 'imports/hook.events.createProduct.failed.json';
+
+    public const CREATE_DATADUMP_DIRECTORY = 'imports/products/createProduct';
+    public const CREATE_DATADUMP_SUCCESS_PRODUCT = 'moduleFunction.ShopModule::naturalSearchShopFlatProductForHooks.success.json';
+    public const CREATE_DATADUMP_FAIL_PRODUCT = 'moduleFunction.ShopModule::naturalSearchShopFlatProductForHooks.failed.json';
+
+    public function dataProviderTestImportWithStatusReporting(): array
+    {
+        $tests = [];
+
+        $tests['successful processing'] = [
+            self::CREATE_DATADUMP_SUCCESS_HOOK,
+            1,
+            1,
+            AbstractProductImport::SYNC_STATUS_SUCCESS,
+            8,
+        ];
+
+        $tests['failed processing'] = [
+            self::CREATE_DATADUMP_FAILED_HOOK,
+            0,
+            1,
+            AbstractProductImport::SYNC_STATUS_FAILED,
+            9,
+        ];
+
+        return $tests;
+    }
+
+    /**
+     * @dataProvider dataProviderTestImportWithStatusReporting
+     *
+     * @throws Throwable
+     */
+    public function testImportWithStatusReporting(
+        string $dumpHookFile,
+        int $expectedProductCount,
+        int $expectedStatusCallCount,
+        string $expectedStatus,
+        int $expectedShopProductId
+    ): void {
+        $setShopProductObjectSyncStatusForHookCallCount = 0;
+
+        StoreKeeperApi::$mockAdapter
+            ->withModule(
+                'ShopModule',
+                function (MockInterface $module) use (
+                    $expectedStatus,
+                    $expectedShopProductId,
+                    &$setShopProductObjectSyncStatusForHookCallCount
+                ) {
+                    $module->allows('setShopProductObjectSyncStatusForHook')
+                        ->andReturnUsing(function ($got) use (
+                            $expectedStatus,
+                            $expectedShopProductId,
+                            &$setShopProductObjectSyncStatusForHookCallCount
+                        ) {
+                            $request = $got[0];
+                            $this->assertEquals($expectedStatus, $request['status'], 'Reported status is wrong');
+                            $this->assertEquals($expectedShopProductId, $request['shop_product_id'], 'Shop product ID sent is wrong');
```

Comment:
Is it also possible, to assert the extra sent to the API?

Reply:
Unfortunately it would be hard to assert the extras, cause the woocommerce product ID changes every test, as well as date_synchronized and the plugin version as well.